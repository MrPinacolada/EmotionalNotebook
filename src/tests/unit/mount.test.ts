import { describe, it, expect } from 'vitest'
import { createApp } from 'vue'
import { mount } from '@vue/test-utils'
import MajorCalendarComponentVue from '@/components/CalendarModule/MajorCalendarComponent.vue'
import EmotionalCircleCalendarVue from '@/components/CalendarModule/EmotionalCircleCalendar.vue'
import UserPopUpCalendarVue from '@/components/CalendarModule/UserPopUpCalendar.vue'
import MajorHeaderComponentVue from '@/components/HeaderModule/MajorHeaderComponent.vue'
import MajorLeftBarComponentVue from '@/components/LeftBarModule/MajorLeftBarComponent.vue'
import AppVue from '@/App.vue'
describe('test itself', () => {
  it('test itself', () => {
    expect(1 + 1).toBe(2)
  })
})
describe('mountTest', () => {
  it('test App', () => {
    const app = createApp(AppVue)
    const wrapper = mount(app)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('test major view', () => {
    const app = createApp(MajorCalendarComponentVue)
    const wrapper = mount(app)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('test custom component', () => {
    const app = createApp(EmotionalCircleCalendarVue)
    const wrapper = mount(app)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('test custom component', () => {
    const app = createApp(UserPopUpCalendarVue)
    const wrapper = mount(app)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('test custom component', () => {
    const app = createApp(MajorHeaderComponentVue)
    const wrapper = mount(app)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('test custom component', () => {
    const app = createApp(MajorLeftBarComponentVue)
    const wrapper = mount(app)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
