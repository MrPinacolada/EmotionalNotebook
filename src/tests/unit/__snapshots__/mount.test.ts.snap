// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mountTest > test App 1`] = `""`;

exports[`mountTest > test custom component 1`] = `""`;

exports[`mountTest > test custom component 2`] = `""`;

exports[`mountTest > test custom component 3`] = `""`;

exports[`mountTest > test custom component 4`] = `""`;

exports[`mountTest > test major view 1`] = `""`;
